src/test/kotlin/uk/gov/dluhc/emsintegrationapi/database/repository/PostalVoteApplicationRepositoryIntegrationTest.kt

```diff
@@ -56,4 +57,46 @@ class PostalVoteApplicationRepositoryIntegrationTest : AbstractRepositoryIntegra
             }
         }
     }
+
+    @Test
+    fun `should return records by gss codes and record status order by created date`() {
+        // Given
+        val approvalDetailsValid = buildApprovalDetailsEntity(gssCode = GSS_CODE_1)
+        val approvalDetailsInvalid = buildApprovalDetailsEntity(gssCode = "5678")
+        val listOfApplications =
+            IntStream.rangeClosed(1, 30).mapToObj {
+                val approvalDetails = if (it % 2 == 0) {
+                    approvalDetailsValid
+                } else {
+                    approvalDetailsInvalid
+                }
+                buildPostalVoteApplication(applicationId = it.toString(), approvalDetails = approvalDetails)
+            }.toList()
+
+        postalVoteApplicationRepository.saveAllAndFlush(listOfApplications)
+
+        // When
+        val applicationsReceived =
+            postalVoteApplicationRepository.findByApprovalDetailsGssCodeInAndStatusOrderByDateCreated(
+                listOf(GSS_CODE_1, "9010"),
+                RecordStatus.RECEIVED,
+                Pageable.ofSize(10)
+            )
+
+        // That
+        val numberOfApplicationsReceived = applicationsReceived.size
+        assertThat(numberOfApplicationsReceived).isLessThan(11)
```

Comment:
@inderjitvirdi suggested to assert the exact size, instead of less than 11, it should be 10